lib/Wikia/src/Service/User/Auth/MediaWikiAuthService.php

```diff
@@ -2,8 +2,66 @@
 
 namespace Wikia\Service\User\Auth;
 
+use Wikia\Service\Helios\ClientException;
+use Wikia\Service\Helios\HeliosClient;
+use Wikia\Logger\Loggable;
+
 class MediaWikiAuthService implements AuthService {
+	use Loggable;
+
+	private $heliosClient;
+
+	/**
+	 * @Inject({Wikia\Service\Helios\HeliosClient::class})
+	 */
+	public function __construct( HeliosClient $client ) {
+		$this->heliosClient = $client;
+	}
+
+	/**
+	 * @param HeliosClient $client
+	 */
+	public function setHeliosClient( HeliosClient $client ) {
+		$this->heliosClient = $client;
+	}
+
+	public function authenticate( $username, $password ): AuthResult {
+		$this->info( 'HELIOS_LOGIN authenticate', [ 'method' => __METHOD__, 'username' => $username ] );
+
+		$result = false;
+		$authMethod = AuthResult::AUTH_TYPE_FAILED;
+		$status = \WikiaResponse::RESPONSE_CODE_ERROR;
 
+		try {
+			list( $status, $loginInfo ) = $this->heliosClient->login( $username, $password );
+			if ( !empty( $loginInfo->error ) ) {
+				if ( $loginInfo->error === 'access_denied' ) {
+					$this->info(
+						'HELIOS_LOGIN authentication_failed',
+						[ 'response' => $loginInfo, 'username' => $username, 'method' => __METHOD__ ]
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "response" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)